civis_jupyter_notebooks/assets/initialize-git

```diff
@@ -9,13 +9,26 @@ if test $GIT_CREDENTIAL; then
   git config --global user.email "$USER_EMAIL"
   git config --global core.editor "nano"
 
+  if [[ $GIT_HOST = "github.com" ]]; then
+    USERNAME=$GIT_CREDENTIAL
+    PASSWORD=x-oauth-basic
+  elif [[ $GIT_HOST = "bitbucket.org" ]]; then
+    DECODED_CLIENT_SECRET=$(echo $BITBUCKET_CLIENT_SECRET | openssl enc -d -base64)
+    # Exchange the refresh token we have for an actual auth token
+    RESP=$(curl -X POST -u "$BITBUCKET_CLIENT_ID:$DECODED_CLIENT_SECRET" https://bitbucket.org/site/oauth2/access_token -d grant_type=refresh_token -d refresh_token=$GIT_CREDENTIAL)
+    REPO_HTTP_TOKEN=$(echo $RESP | awk '{print $2}' | sed 's|[",]||g')
+
+    USERNAME=x-token-auth
+    PASSWORD=$REPO_HTTP_TOKEN
+  fi
```

Comment:
How did you test that this all works?

Reply:
I have a [civis-jupyter-python3 dev branch](https://github.com/civisanalytics/civis-jupyter-python3/pull/29) which installs this [civis-jupyter-notebook dev branch](add-bitbucket-to-init-git). Then I created a [notebook in staging](https://platform-staging.civisanalytics.com/spa/#/notebooks/6072) which points to the docker image that was built. It had to be tested in staging because we also need these ENV variables which are in console - https://github.com/civisanalytics/console/pull/12728. 
<img width="779" alt="Screen Shot 2019-04-24 at 11 27 27 AM" src="https://user-images.githubusercontent.com/29211765/56676473-fe04ff00-6683-11e9-826c-75843d905372.png">
.

Reply:
Also, there is similar bitbucket logic in the replicator repo - https://github.com/civisanalytics/replicator/blob/master/git-clone.sh